### PR TITLE
Fix #485 improve smarty tpl and Home while there is no data to show

### LINKED ISSUE #485: new-theme template must be resilient when no data
when there is no data on some widget, smarty template should be resilient

while working on #473 on local instance of new-theme:

- `footer.tpl` seems broken as I dont have the rest / navbar...

````
Argument 1 passed to smarty_modifier_print_date() must be an instance of DateTime, null given, called in /tmp/smarty/compile/b7266d1fbbc231892001e0fe1584d1188e463762_2.file.news.tpl.php on line 64 [/var/www/html/app-templates/smarty/plugins/modifier.print_date.php:14]
````
l64 is
````
<?php echo smarty_modifier_print_date($_smarty_tpl->tpl_vars['news']->value->created_on_datetime);?>
````
smarty_modifier_print_date seems not accepting null as Date :)

update modifiers to be more resilient with no data

### PR DESCRIPTION
**Pull request by boly38**
new-theme template must be resilient when no data

- Fix #485 improve smarty tpl and Home while there is no data to show

@kumy don't hesitate to rock my suggest if we could simplify / I dont know very well tpl feature (ex. multiple if conditon, function overrides...)

### CODE REVIEW

**Comment by kumy** [comment]
Will check later, thank for that !
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: website/app-templates/smarty/elements/move_as_list.tpl [hunk 1]
```diff
@@ -16,5 +16,5 @@
         <br />
         <small>{$move->author|userlink:$move->username nofilter}</small>
     </td>
-    <td class="text-right">{if $move->logtype->isCountingKilometers()}{$move->distance} km{/if}</td>
+    <td class="text-right">{if $move->logtype}{if $move->logtype->isCountingKilometers()}{$move->distance} km{/if}{/if}</td>
```
Comment (kumy) [inline website/app-templates/smarty/elements/move_as_list.tpl]: One if is sufficient. `{If test && test->func()}{/if}`

File: website/app-templates/smarty/pages/user_recent_moves.tpl [hunk 2]
```diff
@@ -24,7 +24,7 @@
                 </thead>
                 <tbody>
                     {foreach from=$moves.subset item=item}
```
Comment (kumy) [inline website/app-templates/smarty/pages/user_recent_moves.tpl]: I ve seen that case elsewhere and opened an issue on `ikez/cortex` repo about it. IIRC we solved it differently. Grep for subset in other templates

File: website/app-templates/smarty/plugins/modifier.cachelink.php [hunk 3]
```diff
@@ -11,7 +11,11 @@
  * Purpose:  outputs a cache link
  * -------------------------------------------------------------
  */
-function smarty_modifier_cachelink(\GeoKrety\Model\Move $move) {
+function smarty_modifier_cachelink($move) {
+    if ($move == null || !$move instanceof\GeoKrety\Model\Move) {
```
Comment (kumy) [inline website/app-templates/smarty/plugins/modifier.cachelink.php]: Why not try `smarty_modifier_cachelink(?\GeoKrety\Model\Move $move` or `smarty_modifier_cachelink(\GeoKrety\Model\Move $move = null` ;)

File: website/app-templates/smarty/pages/user_recent_moves.tpl (hunk 2 above)
Comment (kumy) [inline website/app-templates/smarty/pages/user_recent_moves.tpl]: https://github.com/ikkez/f3-cortex/issues/87#issuecomment-523035984

File: website/app/GeoKrety/Controller/Home.php [hunk 4]
```diff
@@ -13,8 +13,10 @@ public function get($f3) {
         // Load statistics
         $siteStats = new SiteStats();
         $result = $siteStats->find(array('name LIKE ?', 'stat_%'));
-        foreach ($result as $item) {
-            $statistics[$item['name']] = $item['value'];
+        if ($result != null) {
+            foreach ($result as $item) {
```
Comment (kumy) [inline website/app/GeoKrety/Controller/Home.php]: `foreach($mapper->foo?:[] as $item)`

**Comment by kumy** [comment]
Not easy to review on mobile phone. I really hate that! (Where is my keyboard,? Argh to much boxes around ;))

File: website/app-templates/smarty/plugins/modifier.cachelink.php (hunk 3 above)
Comment (boly38) [reply website/app-templates/smarty/plugins/modifier.cachelink.php]: thoses 2 ways results in 

    Argument 1 passed to smarty_modifier_cachelink() must be an instance of GeoKrety\Model\Move or null, bool given

File: website/app-templates/smarty/elements/move_as_list.tpl (hunk 1 above)
Comment (boly38) [reply website/app-templates/smarty/elements/move_as_list.tpl]: yep

File: website/app-templates/smarty/pages/user_recent_moves.tpl (hunk 2 above)
Comment (boly38) [reply website/app-templates/smarty/pages/user_recent_moves.tpl]: I dont remember why I update this user dedicated tpl, revert my upd

**Comment by kumy** [comment]
Then error in parent template, null should arrive there, no?
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: website/app-templates/smarty/plugins/modifier.cachelink.php (hunk 3 above)
Comment (boly38) [reply website/app-templates/smarty/plugins/modifier.cachelink.php]: > Then error in parent template, null should arrive there, no?

Yep, I updated parent template to not show empty list entries => `(bool(false))` in fact..

So I will revert instanceof ;)

**Comment by boly38** [comment]
navbar / news link > must we display navbar news link if and only if we have news in db ?

just general rmk about nt/
- minor / I've never seen "trous-de-taupes" idea.. amazing :)
- really impressed by amount of work u've done

**Comment by boly38** [comment]
> I don't like those `if ($geokret == null) {` and `if ($move == null ) {`. Where do they fail? Template couldn't be improved?

U're right, with parent template fix, this check are no more usefull, I revert them. I will squash

File: website/app-templates/smarty/pages/home.tpl [hunk 5]
```diff
@@ -38,10 +41,14 @@
     </tbody>
   </table>
 </div>
+{else}{* no recent moves means db is empty *}
```
Comment (kumy) [inline website/app-templates/smarty/pages/home.tpl]: If `else` is not used please remove, else add a message targeted to the developer.

**Comment by kumy** [comment]
ho, please fix travis errors first!

**Comment by boly38** [comment]
> ho, please fix travis errors first! (Even if they're not your fault ;))
> 
> Edit: if travis is not ok, then no translation strings are exported to crowdin

fixed phpcs, and now I got `Unable to connect to apt.cache.travis-ci.com:http:`

**Comment by kumy** [comment]
I've restarted the build, hope it was a temporary issue

File: website/app-templates/smarty/navbar.tpl [hunk 6]
```diff
@@ -31,7 +31,9 @@
 
             <ul class="nav navbar-nav">
                 <li><a href="{'home'|alias}"><span class="glyphicon glyphicon-home" aria-hidden="true"></span> {t}Home{/t}</a></li>
+                {if $news}
```
Comment (kumy) [inline website/app-templates/smarty/navbar.tpl]: I did not thought about that before, but in this context, `$news` only exists in some pages (`home` at least). We should revert this, and add proper message in `news_list.tpl` for the case where there is no news to list.

Comment (boly38) [reply website/app-templates/smarty/navbar.tpl]: okay 😀

Comment (kumy) [reply website/app-templates/smarty/navbar.tpl]: (I've already prepared a fix for it)

**Pull request merged**